worlds/sc2/Locations.py

```diff
@@ -78,7 +78,8 @@ def get_locations(multiworld: Optional[MultiWorld], player: Optional[int]) -> Tu
     # Note: rules which are ended with or True are rules identified as needed later when restricted units is an option
     logic_level = get_option_value(multiworld, player, 'required_tactics')
     adv_tactics = logic_level != RequiredTactics.option_standard
-    kerriganless = get_option_value(multiworld, player, 'kerrigan_presence') not in kerrigan_unit_available
+    kerriganless = get_option_value(multiworld, player, 'kerrigan_presence') not in kerrigan_unit_available \
+        or SC2Campaign.HOTS not in get_enabled_campaigns(multiworld, player)
```

Comment:
These checks are no longer necessary if the slot_data was updated earlier.

Reply:
I believe slot data occurs after generation?  So it wouldn't be available at that point.